public/app/features/alerting/unified/state/actions.ts

```diff
@@ -70,62 +83,161 @@ export function fetchAllPromAndRulerRulesAction(force = false): ThunkResult<void
   };
 }
 
-export function deleteRuleAction(ruleLocation: RuleLocation): ThunkResult<void> {
-  /*
-   * fetch the rules group from backend, delete group if it is found and+
-   * reload ruler rules
-   */
-  return async (dispatch) => {
-    const { namespace, groupName, ruleSourceName, ruleHash } = ruleLocation;
-    //const group = await fetchRulerRulesGroup(ruleSourceName, namespace, groupName);
-    const groups = await fetchRulerRulesNamespace(ruleSourceName, namespace);
-    const group = groups.find((group) => group.name === groupName);
-    if (!group) {
-      throw new Error('Failed to delete rule: group not found.');
+async function findExistingRule(ruleIdentifier: RuleIdentifier): Promise<RuleWithLocation | null> {
+  if (isGrafanaRuleIdentifier(ruleIdentifier)) {
```

Comment:
nit: maybe introduce early exit instead of having `if`..`else` to prevent nesting.

Reply:
I'm still readjusting habits from my previous job, where early exits where frowned upon

- [ ] early exit

Reply:
in this particular case there is a split between two branches; no option for an early exit